### ISSUE TITLE
imresamplemex.cpp unintialized variable bug

### ISSUE THREAD

**Issue by JoeyZhu** [pdollar/toolbox#13]
resample output has wild big value in some case, due to use uninitialized variable  in imresamplemex.cpp:

    for(; y<hn; y++) { B0[ybs[y]] += C[yas[y]] * ywts[y];

should initialize B0 before use:

    for(int i = 0; i < hb; i++) B0[i] = 0;

**Comment by hardegg** [comment]
Yes it's true. I'm not using matlab but a pure c++ compiling. Inside function resample, I add a line in the beginning and everything is ok now.
  memset(B, 0, hb*wb*sizeof(T));

I guess it might be ok for matlab. Maybe before passing to function the variable will be initialized to zeros by default. However, in c++ it's not true.

Hope the author can fix this.
